From 18fe0f0b6c645b314d49b377602cc681167422e6 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 22 Sep 2023 11:11:27 +0200
Subject: [PATCH 22/30] Repair coverage to be compatible with 8.3 test env

Impacted drivers: LINSTOR, MooseFS and ZFS.
- Ignore all linstor.* members during coverage,
  the module is not installed in github runner.
- Use mock from unittest, the old one is not found now.
- Remove useless return from LinstorSR scan method.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py    | 3 +--
 tests/pylintrc          | 2 +-
 tests/test_MooseFSSR.py | 4 +++-
 tests/test_ZFSSR.py     | 4 ++--
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 1c9bd54a..fe6d01d4 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -779,9 +779,8 @@ class LinstorSR(SR.SR):
 
         # Update the database before the restart of the GC to avoid
         # bad sync in the process if new VDIs have been introduced.
-        ret = super(LinstorSR, self).scan(self.uuid)
+        super(LinstorSR, self).scan(self.uuid)
         self._kick_gc()
-        return ret
 
     @_locked_load
     def vdi(self, uuid):
diff --git a/tests/pylintrc b/tests/pylintrc
index a982913b..4588675b 100644
--- a/tests/pylintrc
+++ b/tests/pylintrc
@@ -84,7 +84,7 @@ ignored-classes=SQLObject
 
 # List of members which are set dynamically and missed by pylint inference
 # system, and so shouldn't trigger E0201 when accessed.
-generated-members=REQUEST,acl_users,aq_parent
+generated-members=REQUEST,acl_users,aq_parent,linstor.*
 
 # List of module names for which member attributes should not be checked
 # # (useful for modules/projects where namespaces are manipulated during runtime
diff --git a/tests/test_MooseFSSR.py b/tests/test_MooseFSSR.py
index feaac626..f4e0a852 100644
--- a/tests/test_MooseFSSR.py
+++ b/tests/test_MooseFSSR.py
@@ -1,4 +1,6 @@
-import mock
+from unittest import mock
+import unittest
+
 import MooseFSSR
 import unittest
 
diff --git a/tests/test_ZFSSR.py b/tests/test_ZFSSR.py
index d0cca935..544ea39a 100644
--- a/tests/test_ZFSSR.py
+++ b/tests/test_ZFSSR.py
@@ -1,10 +1,10 @@
+from unittest import mock
+import unittest
 import uuid
 
 import FileSR
-import mock
 import os
 import SR
-import unittest
 import ZFSSR
 
 
